reference_data/management/tests/update_gencode_tests.py

```diff
@@ -0,0 +1,193 @@
+
+        # Test missing genome_version when a GTF file is provided
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp')
+        self.assertEqual(ce.exception.message, "The genome version must also be specified after the gencode GTF file path")
+
+        # Test gencode_release and genome_version mis-matched case 1
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp.gz', '38')
+        mock_isfile.assert_called_with('/var/tmp.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version: 38. gencode v19 only has a GRCh37 version")
+
+        # Test gencode_release and genome_version mis-matched case 2
+        mock_isfile.reset_mock()
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=20', '/var/tmp1.gz', '37')
+        mock_isfile.assert_called_with('/var/tmp1.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version: 37. gencode v20, v21, v22 only have a GRCh38 version")
+
+        # Test genome_version != 38 requires lifted data
+        mock_isfile.reset_mock()
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=23', '/var/tmp2.gz', '37')
+        mock_isfile.assert_called_with('/var/tmp2.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version for file: /var/tmp2.gz. gencode v23 and up must have 'lift' in the filename or genome_version arg must be GRCh38")
+
+    @mock.patch('reference_data.management.commands.update_gencode.logger')
+    def test_update_gencode_command_bad_gtf_data(self, mock_logger):
+        # Test wrong number data feilds in a line
+        temp_bad_file_path = os.path.join(self.test_dir, 'bad.gencode.v23lift37.annotation.gtf.gz')
+        with gzip.open(temp_bad_file_path, 'w') as f:
+            f.write(u''.join(BAD_FIELDS_GTF_DATA))
+        with self.assertRaises(ValueError) as ve:
+            call_command('update_gencode', '--gencode-release=23', temp_bad_file_path, '37')
+        self.assertEqual(ve.exception.message, 'Unexpected number of fields on line #0: [\'gene\', \'11869\', \'14412\', \'.\', \'+\', \'.\', \'gene_id "ENSG00000223972.4";\']')
+        mock_logger.info.assert_called_with('Loading {} (genome version: 37)'.format(temp_bad_file_path))
+
+    @mock.patch('reference_data.management.commands.update_gencode.logger')
+    @mock.patch('reference_data.management.commands.update_gencode.download_file')
+    def test_update_gencode_command_url_generation(self, mock_download, mock_logger):
+        # Test the code paths of generating urls, gencode_release == 19, and --reset option
```

Comment:
can you update this comment now that you moved --reset?

Reply:
Okay